docs/interface/sparse.md

```diff
@@ -1,38 +1,139 @@
+they are assumed to take some default value (usually zero) and storage for these fields does not 
+need to be allocated in those sub-regions. Fields that are explicitly evolved on only certain 
+sub-regions of the grid are termed "sparse fields." 
+
+Some examples of where sparse fields might be desirable: 
+- Properties (density, temperature, etc.) of materials that are only present in certain parts of 
+  the simulation. One does not want to allocate memory that contains all zeros and perform 
+  operations on zeros in regions where the material is not present.
+- A self-gravitating fluid in a vacuum. Often, astrophysics codes will model this by including 
+  fluid throughout the entire simulation domain but set a floor on the density and temperature of 
+  the material to create an effective "atmosphere" around the actual self-gravitating system. 
+  Regions that are set to the floor do not necessarily need to be explicitly tracked and the fluid 
+  quantities could be treated as sparse fields. 
+- Material that is transitioning between non-equilibrium nuclear burning and nuclear statistical 
+  equilibrium (NSE). In the non-equilibrium regions, the abundances of all nuclear species are 
+  required to define the composition of the material while in the regions that are in NSE only one 
+  field (the electron fraction) is required to define the composition. 
+
+In Parthenon, sparse fields are implemented at the block level, meaning that a field is allocated 
+everywhere or nowhere on each block in the simulation. The set of possible operations for a sparse 
+field are:
+
+- *Allocation on initialization:* Sparse variables need to be allocated by hand in the 
+  `ProblemGenerator` in regions where they are non-zero.  
+- *Allocation due to local changes:* For some types of sparse fields (e.g. nuclear abundances fields 
+  for material transitioning in and out of NSE), it may be desirable to allocate them based on the 
+  local state on the block (e.g. allocate abundances if the temperature is anywhere on the block 
+  below some threshold). This is an operation that needs to be defined in a task by the downstream 
+  code, since it is problem specific. **Currently, Parthenon does not have this capability. It will 
+  probably be necessary to define some hooks for checking blocks for the necessity of allocation and 
+  having some internal Parthenon functions for actually performing the allocation.**
+- *Allocation due to neighbors:* If a sparse field on a neighbor block passes ghost data that is 
+  anywhere above the `allocation_threshold`, the receiving block will allocate the sparse field if 
+  it is unallocated before loading the ghost data. `allocation_threshold` is set in the `Metadata` 
+  of a field.  A detailed description of sparse ghost zone communication is 
+  given [here](../sparse_boundary_communication.md). 
+- *Allocation due to other fields:* In some instances, we may want to allocate or deallocate a 
```

Comment:
I think that it should be `or` between `allocation due to neighbors` and `allocation due to other fields`. We should always default to allocating a field if anything requires allocation, since that won't degrade the accuracy of the calculation.

Right now, there is no mechanism for `allocation due to other fields` in the code.

Reply:
> Right now, there is no mechanism for `allocation due to other fields` in the code.

Now I'm confused. I thought this PR is about adding this functionality (as described also here in the doc). What am I missing?

Reply:
Sorry, there are two ideas here. The first is that you would like to always allocate one field whenever some other field is allocated. This is what this PR introduced. The second, which is not implemented, is allocating a field based upon some more complicated criterion in the interior of a block (e.g. the temperature being above/below some threshold).